business/authentication/openid_auth_controller.go

```diff
@@ -361,6 +333,14 @@ func (c OpenIdAuthController) redirectToAuthServerHandler(w http.ResponseWriter,
 		return
 	}
 
+	// Kiali only supports the authorization code flow.
+	if !isOpenIdCodeFlowPossible() {
+		w.Header().Set("Content-Type", "text/plain")
+		w.WriteHeader(http.StatusConflict)
```

Comment:
I'm not sure this is a 409. There's no way in which the user can change the request in order to fix this. In order to fix this they would either need to change their auth strategy or move to a provider that supports the code flow right? Maybe a `500` error or 501: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501?

Reply:
I think I agree with changing this to `501`

Reply:
Fixed.